doc/pserver.md

```diff
@@ -0,0 +1,226 @@
+# Parameter Server Design
+
+
+## Overview
+
+
+## PServer
+
+
+### Compoments
+
+
+![pserver](./images/pserver.png)
+
+
+PServer contains three main compoments:
+
+- KVStore
+- Gradient Queue
+- Optimizer
+
+1. The worker initializes a model, and pushes parameters to KVStore.
```

Comment:
How do we choose which worker does this initialization job?

Reply:
push **gradients**?

Reply:
Currently, all workers will do the initialization job, and push parameters to KVStore. If the KVStore has the parameter already, it will discard the later parameter.